nixos/modules/services/networking/cloudflare-warp.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cloudflare-warp;
+
+in {
+  meta.maintainers = with maintainers; [ wolfangaukang ];
+
+  options = {
+    services.cloudflare-warp = {
+      enable = mkEnableOption "cloudflare-warp, a service that replaces the connection between your device and the Internet with a modern, optimized, protocol";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = with pkgs; [ cloudflare-warp ];
+
+    systemd = {
+      packages = [ pkgs.cloudflare-warp ];
+      services.warp-svc = {
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          StateDirectory = "cloudflare-warp";
```

Comment:
Not to great to have no hardening at all.

Reply:
What do you mean? I didn't get it.

Reply:
> What do you mean? I didn't get it.

Giving just enough permission for it to do its work so no compromise if it misbehaves

Reply:
Now I get it, I just checked some examples of modules and now it's clear of what is expected.